graph/schema.graphqls

```diff
@@ -1,69 +1,201 @@
+"""
 type Query {
+  """
+  Search for resources and their relationships. 
+  *[PLACEHOLDER] Results only include kubernetes resources for which the authenticated user has list permission.*
+
+  For more information see the feature spec.
+  """
   search(input: [SearchInput]): [SearchResult]
-  messages : [Message]
+  
+  """
+  Query all values for the given property.  
+  Optionally, a query can be included to filter the results.  
+  For example, if we want to get the names of all resources in the namespace foo, we can pass a query with the filter `{property: namespace, values:['foo']}`
+  
+  **Default limit is** 10,000  
+  A value of -1 will remove the limit. Use carefully because it may impact the service.
+  """
+  searchComplete(property: String!, query: SearchInput, limit: Int): [String]
+
+  """
+  Returns all properties from resources currently in the index.
+  """
   searchSchema: Map
+
+  """
+  Saved search queries for the authenticated user.
+  **[PLACEHOLDER] This query is not yet implemented in V2.**
+  """
   savedSearches: [userSearch]
-  searchComplete(property: String!, query: SearchInput, limit: Int): [String]
 
+  """
+  Additional information about the service status or conditions found while processing the query.  
+  This is similar to the errors query, but without implying that there was a problem processing the query.
+  """
+  messages: [Message]
 }
 
-# This is a mutation we will be doing
+"""
+Supported mutations
+"""
 type Mutation {
-    # Delete search query for the current user.
+    """
+    Delete search query for the authenticated user.
+    **[PLACEHOLDER] This query is not yet implemented in V2.**
+    """
     deleteSearch(resource: String): String
-    # Save a search query for the current user.
+    """
+    Save a search query for the authenticated user.
+    **[PLACEHOLDER] This query is not yet implemented in V2.**
+    """
     saveSearch(resource: String): String
 }
 
+
+"""
+Defines a key/value to filter results.  
+When multiple values are provided for a property, it is interpreted as an OR operation.
+"""
 input SearchFilter {
+    """
+    Name of the property (key).
+    """
     property: String!
+    """
+    Values for the property. Multiple values per property are interpreted as an OR operation.
+    Optionally one of these operations `=,!,!=,>,>=,<,<=` can be included at the begining of the value.
+    By default the equality operation is used.
+    """
     values: [String]!
   }
+
+
+"""
+Input options to the search query.
+"""
 input SearchInput {
+    """
+    List of strings to match resources.  
+    Will match resources containing any of the keywords in any text field.  
+    When multiple keywords are provided, it is interpreted as an AND operation.  
+    Matches are case insensitive.
+    """
     keywords: [String]
+
+    """
+    List of SearchFilter, which is a key(property) and values.  
+    When multiple filters are provided, results will match all fiters (AND operation).
```

Comment:
Should we add the values available for datetime fields here, to satisfy the 3rd acceptance criteria in [#20787](https://github.com/stolostron/backlog/issues/20787)?

The values available for datetime fields (Ex: `created`, `startedAt`) are `hour`, `day`, `week`, `month`, `year`.

Reply:
Maybe the `input SearchFilter{` definition is a better place. (around line 75)